### ISSUE TITLE
AlphaHPP without 3D Secure (`3DSecureBypass` = `Y`) and with `Currency` = `AUD`: «Provider unable to process request» / «"ProviderName": "westpac"»

### ISSUE THREAD

**Issue by dmitrii-fediuk** [mage2pro/alphacommercehub#12]
This isssue is related to the issue #8: ([AlphaHPP without 3D Secure (`3DSecureBypass` = `Y`): «Provider unable to process request» / «"ProviderName": "westpac"»](https://github.com/mage2pro/alphacommercehub/issues/8)).

I have got an [answer](https://github.com/mage2pro/alphacommercehub/issues/8#issuecomment-343819520) from the AlphaCommerceHub team for the issue #8:

> The test merchant is going to Westpac test host which only supports `AUD` and `NZD`.
`USD` was submitted. Please **change to `AUD`** and try again

So from now on, I use the Australian dollar (`AUD`) for all my payment requests. 
But it does not help me:

![01](https://user-images.githubusercontent.com/83204/32734333-cb38ca32-c8a2-11e7-81a1-821585503a4c.png)

![02](https://user-images.githubusercontent.com/83204/32734340-d086b9ea-c8a2-11e7-934e-0c878921ace7.png)

### General
```
Request URL: https://hubuat.alphacommercehub.com.au/pp/ac580978-81f0-4716-bfc3-3bc46eab8603
Request Method: POST
Status Code: 200 OK
Remote Address: 76.12.61.237:443
Referrer Policy: no-referrer-when-downgrade
```

### Response headers
```
HTTP/1.1 200 OK
Date: Mon, 13 Nov 2017 15:40:34 GMT
Server: Microsoft-IIS/8.5
Cache-Control: private
Content-Type: text/html; charset=utf-8
X-AspNetMvc-Version: 4.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Vary: Accept-Encoding
Content-Encoding: gzip
Content-Length: 7174
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
```

### Request headers
```
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9,ru;q=0.8
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 1126
Content-Type: application/x-www-form-urlencoded
Cookie: __RequestVerificationToken=n-B5epNpZExPBpdLRWY_IOg_zxlnxkKjL1utioq1JrxE6rgCQN9BJ480lFpp8mWnGdzLeRMhjjnUk-PHF_BJBywBGWw7qNJYr6zSdMzZPcjvYobCiCRY5Eyc2vjI9cd0Pdv0_Q2; _timeZoneOffset=180; _ga=GA1.3.685466192.1508711293; ASP.NET_SessionId=gam2fam11wnk5cryudsqbxps; .ASPXAUTH=3ED353E66D8A6BF8D78C2CC3EE304A9AC0F02B9391057D31ACE1A74CE33F5D7FC2872FF60000802C6BF713AEC8EC7835F816BA5120B2A0521E45C5521D28A5AF78B5C838794351B5DE91E4A8E64BE8344535C633642DA75A90724ED9283621CAA46327BA202B1FB169ED905B653E1DF5F7B9A49DFEA2D150D65D6A87EE13DE1CF0BDF18F77232DF9ADFC3B77652260AD5F77B8A87DF3EC066A0A6A84A96CBDCC7CAB81C410D5E683EA06EEBE0B462A5C8D93DFD0; email=seb%40anterior.com.au
Host: hubuat.alphacommercehub.com.au
Origin: https://localhost.com:900
Pragma: no-cache
Referer: https://localhost.com:900/store/checkout/
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.89 Safari/537.36
```
### Form data
```
3DSecureBypass: Y
Amount: 160706
CancelURL: https://mage2.pro/sandbox/dfe-alphacommercehub/customerReturn
Capture: Y
City: Sydney
Country: AU
Currency: AUD
CustomerID: dfediuk@gmail.com
EmailAddress: dfediuk@gmail.com
FraudCheckBypass: N
IPAddress: 92.243.166.8
Language: en
MerchantDescriptor: Mage2PRO
MerchantID: ANTERIOR001
MerchantTxnID: 1104L725
Method: CC
OrderDetails[0][ItemAmount]: 45554
OrderDetails[0][ItemName]: Alligator Briefcase
OrderDetails[0][ItemQuantity]: 2
OrderDetails[0][ItemTaxExempt]: TRUE
OrderDetails[1][ItemAmount]: 56943
OrderDetails[1][ItemName]: New Very Prive 41 Black
OrderDetails[1][ItemQuantity]: 1
OrderDetails[1][ItemTaxExempt]: TRUE
OrderDetails[2][ItemAmount]: 12654
OrderDetails[2][ItemName]: Shipping
OrderDetails[2][ItemQuantity]: 1
OrderDetails[2][ItemTaxExempt]: TRUE
SocialID: 74041803683
State: Mage2PRO
Street1: 72 Liverpool St
SuccessURL: https://mage2.pro/sandbox/dfe-alphacommercehub/customerReturn
TelNo: 61282688888
TxnDetails: Alligator Briefcase 2 New Very Prive
UserId: 123
Zip: 2000
```

### General
```
Request URL: https://hubuat.alphacommercehub.com.au/pp/auth
Request Method: POST
Status Code: 200 OK
Remote Address: 76.12.61.237:443
Referrer Policy: no-referrer-when-downgrade
```

### Response headers
```
HTTP/1.1 200 OK
Date: Mon, 13 Nov 2017 15:41:00 GMT
Server: Microsoft-IIS/8.5
Cache-Control: private
Content-Type: application/json; charset=utf-8
X-AspNetMvc-Version: 4.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Content-Length: 1504
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
```

### Request headers
```
Accept:*/*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9,ru;q=0.8
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 3021
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
Cookie: __RequestVerificationToken=n-B5epNpZExPBpdLRWY_IOg_zxlnxkKjL1utioq1JrxE6rgCQN9BJ480lFpp8mWnGdzLeRMhjjnUk-PHF_BJBywBGWw7qNJYr6zSdMzZPcjvYobCiCRY5Eyc2vjI9cd0Pdv0_Q2; _timeZoneOffset=180; _ga=GA1.3.685466192.1508711293; ASP.NET_SessionId=gam2fam11wnk5cryudsqbxps; .ASPXAUTH=3ED353E66D8A6BF8D78C2CC3EE304A9AC0F02B9391057D31ACE1A74CE33F5D7FC2872FF60000802C6BF713AEC8EC7835F816BA5120B2A0521E45C5521D28A5AF78B5C838794351B5DE91E4A8E64BE8344535C633642DA75A90724ED9283621CAA46327BA202B1FB169ED905B653E1DF5F7B9A49DFEA2D150D65D6A87EE13DE1CF0BDF18F77232DF9ADFC3B77652260AD5F77B8A87DF3EC066A0A6A84A96CBDCC7CAB81C410D5E683EA06EEBE0B462A5C8D93DFD0; email=seb%40anterior.com.au
Host:hubuat.alphacommercehub.com.au
Origin: https://hubuat.alphacommercehub.com.au
Pragma: no-cache
Referer: https://hubuat.alphacommercehub.com.au/pp/ac580978-81f0-4716-bfc3-3bc46eab8603
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.89 Safari/537.36
X-APC-MerchantTxnID: 1104L725
X-APC-PPSessionId: 7758890d-f71e-4383-ba25-2a96817198b2
X-Requested-With:XMLHttpRequest
```
### Form data
```
cardHolder: DMITRY FEDYUK
cardNumber: 4557 0400 0000 0005
cardExpiryDate: 07 / 21
cardCVV: 123
country: AU
street1: 72 Liverpool St
street2:
city: Sydney
zip: 2000
emailAddress:
token: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJwYXlsb2FkIjoie1wiVXNlcklkXCI6XCIxMjNcIixcIk1lcmNoYW50SURcIjpcIkFOVEVSSU9SMDAxXCIsXCJNZXJjaGFudFR4bklEXCI6XCIxMTA0TDcyNVwiLFwiQ3VycmVuY3lcIjpcIkFVRFwiLFwiQW1vdW50XCI6XCIxNjA3MDZcIixcIlRyYW5zYWN0aW9uVGltZXN0YW1wXCI6bnVsbCxcIk1ldGhvZFwiOlwiQ0NcIixcIkxhbmd1YWdlXCI6XCJlblwiLFwiVHhuRGV0YWlsc1wiOlwiQWxsaWdhdG9yIEJyaWVmY2FzZSAyIE5ldyBWZXJ5IFByaXZlXCIsXCJDYXB0dXJlXCI6XCJZXCIsXCJFbWFpbEFkZHJlc3NcIjpcImRmZWRpdWtAZ21haWwuY29tXCIsXCJJUEFkZHJlc3NcIjpcIjkyLjI0My4xNjYuOFwiLFwiVGVsTm9cIjpcIjYxMjgyNjg4ODg4XCIsXCJDdXN0b21lcklEXCI6XCJkZmVkaXVrQGdtYWlsLmNvbVwiLFwiU29jaWFsSURcIjpcIjc0MDQxODAzNjgzXCIsXCJTdHJlZXQxXCI6XCI3MiBMaXZlcnBvb2wgU3RcIixcIlN0cmVldDJcIjpudWxsLFwiQ2l0eVwiOlwiU3lkbmV5XCIsXCJaaXBcIjpcIjIwMDBcIixcIlN0YXRlXCI6XCJNYWdlMlBST1wiLFwiQ291bnRyeVwiOlwiQVVcIixcIk1lcmNoYW50RGVzY3JpcHRvclwiOlwiTWFnZTJQUk9cIixcIlN1Y2Nlc3NVUkxcIjpcImh0dHBzOi8vbWFnZTIucHJvL3NhbmRib3gvZGZlLWFscGhhY29tbWVyY2VodWIvY3VzdG9tZXJSZXR1cm5cIixcIlRocmVlRFNlY3VyZUJ5cGFzc1wiOlwiWVwiLFwiQ2FyZFR5cGVcIjpcImFtZXgsamNiLG1hc3RlcmNhcmQsZGlzY292ZXIsbWFzdGVyY2FyZGRlYml0LHZpc2EsdmlzYWRlYml0LHVuaW9ucGF5LG1hZXN0cm8sdWF0cCxkaW5lcnNjbHViXCIsXCJGcmF1ZENoZWNrQnlwYXNzXCI6XCJOXCIsXCJJbnN0YWxsbWVudFwiOm51bGwsXCJQYXlMYXRlck1vbnRoc1wiOm51bGwsXCJQYXlOdW1iZXJNb250aHNcIjpudWxsLFwiSW5zdGFsbG1lbnRJRFwiOm51bGwsXCJJbnN0YWxsbWVudFR5cGVcIjpudWxsLFwiQ2FuY2VsVVJMXCI6XCJodHRwczovL21hZ2UyLnByby9zYW5kYm94L2RmZS1hbHBoYWNvbW1lcmNlaHViL2N1c3RvbWVyUmV0dXJuXCIsXCJDYXJkTnVtYmVyXCI6bnVsbCxcIkNhcmRIb2xkZXJcIjpudWxsLFwiQ2FyZEV4cGlyeU1vbnRoXCI6bnVsbCxcIkNhcmRFeHBpcnlZZWFyXCI6bnVsbCxcIkNhcmRDVlZcIjpudWxsLFwiQ2hhbm5lbFR5cGVcIjpcIjA3XCIsXCJTZWN1cmVUb2tlblwiOm51bGwsXCJTZXNzaW9uSWRcIjpudWxsLFwiVGF4QWRkZWRcIjpmYWxzZSxcIlRheFJhdGVcIjowLjAsXCJOb3RpZmljYXRpb25FbWFpbFwiOm51bGwsXCJSZXR1cm5UZXh0XCI6bnVsbCxcIlJlcGx5VXJsXCI6bnVsbCxcIlVwZGF0ZVVSTFwiOm51bGwsXCJPcmlnaW5cIjpcIkhQUFwiLFwiVHJhbnNhY3Rpb25UeXBlXCI6bnVsbCxcIkN1c3RvbTFcIjpudWxsLFwiUGF5V2l0aFBvaW50c1wiOm51bGwsXCJNZXJjaGFudERhdGFcIjpudWxsLFwiVGVybWluYWxcIjpudWxsLFwiUGF5bWVudFByb2Nlc3NvckFQSUlkXCI6bnVsbCxcIk9yZGVyRGV0YWlsc1wiOlt7XCJJdGVtTmFtZVwiOm51bGwsXCJJdGVtUXVhbnRpdHlcIjowLFwiSXRlbUFtb3VudFwiOm51bGwsXCJJdGVtUHJpY2VcIjpudWxsLFwiSXRlbVRheEV4ZW1wdFwiOmZhbHNlfSx7XCJJdGVtTmFtZVwiOm51bGwsXCJJdGVtUXVhbnRpdHlcIjowLFwiSXRlbUFtb3VudFwiOm51bGwsXCJJdGVtUHJpY2VcIjpudWxsLFwiSXRlbVRheEV4ZW1wdFwiOmZhbHNlfSx7XCJJdGVtTmFtZVwiOm51bGwsXCJJdGVtUXVhbnRpdHlcIjowLFwiSXRlbUFtb3VudFwiOm51bGwsXCJJdGVtUHJpY2VcIjpudWxsLFwiSXRlbVRheEV4ZW1wdFwiOmZhbHNlfV0sXCJDdXN0b21GaWVsZHNcIjpbXSxcIkFjY2VwdDNEU0F1dGhlbnRpY2F0aW9uVW5hdmFpbGFibGVcIjpmYWxzZSxcIkFjY2VwdDNEU0Vucm9sQnlwYXNzXCI6ZmFsc2UsXCJQUFNlc3Npb25JZFwiOlwiZDA3MTBiY2QtOTlmYy00NTliLWIwMzMtY2VmMDMzZWQ5ODZhXCJ9In0.6Kdcuk52OdTZefk1CLbMFU3GV_pyyn1750MEMfckTbQ
cardExpiryMonth: 7
cardExpiryYear: 2021
payPageID: ac580978-81f0-4716-bfc3-3bc46eab8603
merchantSequenceNumber:
clientIP: 92.243.166.8
```

### Response
```
{
	"Result": {
		"ResponseCode": "1076",
		"ResponseMessage": "Provider unable to process request",
		"ResponseTimestamp": "13112017154101",
		"MerchantID": "ANTERIOR001",
		"MerchantTxnID": "1104L725",
		"PaymentID": "107055879",
		"Currency": "AUD",
		"Amount": "160706",
		"BillCurrency": null,
		"BillAmount": null,
		"CardHolder": "DMITRY FEDYUK",
		"CardType": "VISA",
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"ProviderResps": [
		{
			"ProviderName": "westpac",
			"ProviderOrder": "1",
			"ProviderResponse": "71",
			"ProviderReference": "7001755533",
			"ResponseCode": "1076",
			"ResponseTimestamp": "14112017024101"
		}
	],
	"Country": {
		"CountryName": null,
		"CountryCode": null
	},
	"MethodResult": {
		"AuthCode": null,
		"CVVResult": "M",
		"AVSStreetResult": null,
		"AVSZipResult": null,
		"SecureToken": null,
		"Status": "1",
		"AmountReceived": null,
		"RedirectUrl": null,
		"FeeAmount": null,
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null,
		"ACSUrl": null,
		"PAReq": null
	},
	"ThreeDSecure": {
		"AuthenticationValue": null,
		"AuthenticationStatus": null,
		"Enrolled": null,
		"XID": null,
		"ECI": null,
		"PARes": null,
		"CAVVAlgorithm": null,
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"TokenResult": {
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"PaymentInfo": {
		"PaymentBrand": "VISA",
		"PaymentBrandType": "CREDIT",
		"PaymentBrandCategory": "BUSINESS",
		"PaymentIssuer": null,
		"PaymentIssuerCountry": "AU",
		"BIN": "455704"
	},
	"FraudResult": {
		"RiskCategory": null,
		"Score": null,
		"Status": null
	},
	"FraudChecks": [],
	"GeoChecks": [],
	"LoyaltyResult": null
}
```

**Comment by dmitrii-fediuk** [comment]
## A «try again»

![1](https://user-images.githubusercontent.com/83204/32755890-6f6a3c4a-c8e8-11e7-8d4f-24595bc07421.png)

```
{
	"Result": {
		"ResponseCode": "1076",
		"ResponseMessage": "Provider unable to process request",
		"ResponseTimestamp": "13112017235037",
		"MerchantID": "ANTERIOR001",
		"MerchantTxnID": "1104L726",
		"PaymentID": "107056437",
		"Currency": "AUD",
		"Amount": "160706",
		"BillCurrency": null,
		"BillAmount": null,
		"CardHolder": "DMITRY FEDYUK",
		"CardType": "MASTERCARD",
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"ProviderResps": [
		{
			"ProviderName": "westpac",
			"ProviderOrder": "1",
			"ProviderResponse": "71",
			"ProviderReference": "7001757533",
			"ResponseCode": "1076",
			"ResponseTimestamp": "14112017105037"
		}
	],
	"Country": {
		"CountryName": null,
		"CountryCode": null
	},
	"MethodResult": {
		"AuthCode": null,
		"CVVResult": "M",
		"AVSStreetResult": null,
		"AVSZipResult": null,
		"SecureToken": null,
		"Status": "1",
		"AmountReceived": null,
		"RedirectUrl": null,
		"FeeAmount": null,
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null,
		"ACSUrl": null,
		"PAReq": null
	},
	"ThreeDSecure": {
		"AuthenticationValue": null,
		"AuthenticationStatus": null,
		"Enrolled": null,
		"XID": null,
		"ECI": null,
		"PARes": null,
		"CAVVAlgorithm": null,
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"TokenResult": {
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"PaymentInfo": {
		"PaymentBrand": "MASTERCARD",
		"PaymentBrandType": "CREDIT",
		"PaymentBrandCategory": "CORPORATE",
		"PaymentIssuer": null,
		"PaymentIssuerCountry": "AU",
		"BIN": "531357"
	},
	"FraudResult": {
		"RiskCategory": null,
		"Score": null,
		"Status": null
	},
	"FraudChecks": [],
	"GeoChecks": [],
	"LoyaltyResult": null
}
```

**Comment by dmitrii-fediuk** [comment]
It is fixed by #17 (The last 3 digits of every payment amount should be «000» in the test mode for Westpac).
